### ISSUE TITLE
read version info from official s3 bucket

### ISSUE THREAD

**Issue by johnnychen94** [johnnychen94/jill.py#18]
There's actually a JSON file in https://julialang-s3.julialang.org/bin/versions.json that hosts all relevant information. Taking advantage of that we can make `update` faster and enabled by default.

cf. https://github.com/JuliaLang/julia/issues/33817

**Comment by crstnbr** [comment]
@johnnychen94 I'm giving this a try (to also fix #54) but be aware I'm not well-versed in python :)

As a start, I replaced `read_releases`, which used to read from the local `releases.csv` file, by this:

```python
def read_releases(stable_only=False) -> List[Tuple]:
    """
    read release info from versions.json (VERSIONS_URL)
    """
    v = json.loads(requests.get(VERSIONS_URL).text)
    releases = []
    for item in v.items():
        ver = item[0]
        stab = item[1]['stable']
        if not stable_only or stab:
            files = item[1]['files']
            for file in files:
                releases.append((ver, file['os'], file['arch']))
    return releases
```

This still creates a `releases` list of tuples of the form `(ver, os, arch)` but compared to the old one
* includes all Julia versions including betas and release candidates and
* does not contain an explicit `latest` entry

However, more importantly, after I made the change I realised that the whole package seems to rely on the idea that there is a local copy of `releases.csv` that can be updated (via `jill update`) etc. My question is now whether we should
* (A) get rid of the local `releases.csv` altogether and always query the remote `versions.json` (sort of what I started above),
* (B) replace the local `releases.csv` by a local copy of `versions.json` which gets updated every once in a while,
* (C) just modify the update mechanism of the current `releases.csv` file based on the remote `versions.json`.

My feeling is that we should go with (A) as
* grabbing and parsing the `versions.json` seems fast enough,
* we would always be up to date,
* maintaining a separate local `releases.csv` seems redundant.
If we went with (B) we would only gain the speed difference between loading from disc vs from the web (which I think isn't too important) and would have to care about syncing the file.

What do you think? If we would go with (A), what needs to change in the package? AFAICT, the whole `jill update` mechanism would be obsolete as its only purpose seems to be to update the local `releases.csv`, correct?

**Comment by johnnychen94** [comment]
I'm so glad that you put some time to take a look at this. I was thinking about path (A) as well, but probably we need some trick to only fetch `versions.json` **once** during one `jill COMMAND` call. Maintaining a separate `releases.csv` and `jill update` functionality becomes redundant to me as well if this gets properly supported, but to make it easier to review the PR, you could still keep the related codes until the final approval.

Mirror sites also provide `versions.json` files, e.g., https://mirrors.bfsu.edu.cn/julia-releases/bin/versions.json. It's okay to hardcode it to https://julialang-s3.julialang.org/bin/versions.json as for now, and leave this mirror-support as future enhancements. (It's not very reliable for Chinese users to fetch aws s3 resources; that's why I built mirror support here)

The unit test might be failed with this change, feel free to modify it if you think it's a reasonable change.
